protege-editor-owl/src/main/java/org/protege/editor/owl/ui/explanation/ExplanationPreferencesGeneralPanel.java

```diff
@@ -15,30 +15,110 @@ public class ExplanationPreferencesGeneralPanel extends OWLPreferencesPanel {
 
 	private static final long serialVersionUID = -3354987384223578780L;
 
+	private JCheckBox checkRecentlyUsed;
+	private SortedPluginsTableModel tableModel;
+
+	@Override
+	public void initialise() throws Exception {
+		setLayout(new BorderLayout());
+		PreferencesLayoutPanel panel = new PreferencesLayoutPanel();
+		add(panel, BorderLayout.NORTH);
+		addInstalledExplanationServicesComponent(panel);
+		addDefaultExplanationServiceComponent(panel);
+		loadFrom(ExplanationPreferences.create().load());
+	}
+
 	@Override
-    public void initialise() throws Exception {
-        setLayout(new BorderLayout());
-        PreferencesLayoutPanel panel = new PreferencesLayoutPanel();
-        add(panel, BorderLayout.NORTH);
-
-        panel.addGroup("Installed explanation services");
-        DefaultListModel<String> pluginModel = new DefaultListModel<>();
-        ExplanationManager manager = new ExplanationManager(getOWLEditorKit());
-        Collection<ExplanationService> services = manager.getExplainers();
-        for (ExplanationService service : services)
-            pluginModel.addElement(service.getName());
-        JList<String> pluginList = new JList<>(pluginModel);
-        pluginList.setToolTipText("Plugins that provide explanation facilities");
-        JScrollPane pluginInfoScrollPane = new JScrollPane(pluginList);
-        pluginInfoScrollPane.setPreferredSize(new Dimension(300, 100));
-        panel.addGroupComponent(pluginInfoScrollPane);
-    }
-
-    @Override
-    public void dispose() throws Exception {
-    }
-
-    @Override
-    public void applyChanges() {
-    }
+	public void dispose() throws Exception {
+		getOWLEditorKit().getModelManager().getExplanationManager().reload();
+	}
+
+	@Override
+	public void applyChanges() {
+		ExplanationPreferences prefs = ExplanationPreferences.create();

```

Comment:
`create()` will erase `defaultExplanationService`, which is not stored in this preferences. One should probably use preferences that have been loaded during `initialize()`